binder/src/test/java/io/grpc/binder/RobolectricBinderSecurityTest.java

```diff
@@ -0,0 +1,425 @@
+
+    @Override
+    public void onCreate() {
+      super.onCreate();
+
+      MethodDescriptor<Empty, Empty> methodDesc = getMethodDescriptor();
+      ServerCallHandler<Empty, Empty> callHandler =
+          ServerCalls.asyncUnaryCall(
+              (req, respObserver) -> {
+                respObserver.onNext(req);
+                respObserver.onCompleted();
+              });
+      ServerMethodDefinition<Empty, Empty> methodDef =
+          ServerMethodDefinition.create(methodDesc, callHandler);
+      ServerServiceDefinition def =
+          ServerServiceDefinition.builder(SERVICE_NAME).addMethod(methodDef).build();
+
+      server =
+          BinderServerBuilder.forAddress(AndroidComponentAddress.forContext(this), binderReceiver)
+              .addService(def)
+              .securityPolicy(
+                  ServerSecurityPolicy.newBuilder()
+                      .servicePolicy(
+                          SERVICE_NAME,
+                          new AsyncSecurityPolicy() {
+                            @Override
+                            ListenableFuture<Status> checkAuthorizationAsync(int uid) {
+                              return Futures.submitAsync(
+                                  () -> {
+                                    SettableFuture<Status> status = SettableFuture.create();
+                                    statusesToSet.add(status);
+                                    return status;
+                                  },
+                                  getExecutor());
+                            }
+                          })
+                      .build())
+              .executor(getExecutor())
+              .scheduledExecutorService(getExecutor())
+              .build();
+      try {
+        server.start();
+      } catch (IOException e) {
+        throw new IllegalStateException(e);
+      }
+
+      Application context = ApplicationProvider.getApplicationContext();
+      ComponentName componentName = new ComponentName(context, SomeService.class);
+      shadowOf(context)
+          .setComponentNameAndServiceForBindService(
+              componentName, checkNotNull(binderReceiver.get()));
+    }
+
+    /**
+     * Returns an {@link ScheduledExecutorService} under which all of the gRPC computations run. The
+     * execution of any pending tasks on this executor can be triggered via {@link #idleLoopers()}.
+     */
+    ScheduledExecutorService getExecutor() {
+      return scheduledExecutorService;
+    }
+
+    void setSecurityPolicyStatusWhenReady(Status status) {
+      getNextEnqueuedStatus().set(status);
+    }
+
+    void setSecurityPolicyFailed(Exception e) {
+      getNextEnqueuedStatus().setException(e);
+    }
+
+    private SettableFuture<Status> getNextEnqueuedStatus() {
+      @Nullable SettableFuture<Status> future = statusesToSet.poll();
+      while (future == null) {
+        // It's possible that either the test thread or the gRPC thread has posted tasks to each
```

Comment:
Is this comment obsolete now that there is just a single thread?

Reply:
Rephrased.

TBH I think we may not even need this loop anymore now that everything runs on the test thread. A single `idleLoopers()` call should take care of everything as there's just one looper.

The test passes that way, but I keep the loop for now to play safe.